### ISSUE TITLE
feat(training): class-balanced sampling para datasets desbalanceados

### ISSUE THREAD

**Issue by Jesus123J** [Jesus123J/sponsorship_mvp#57]
## Problema

Cuando una clase domina el dataset (ej. \`adidas\`: 1000, \`bbva\`: 8), el gradiente del loss queda dominado por la clase mayoritaria. El modelo aprende muy bien adidas y casi ignora bbva. Resultado: muchos falsos positivos del tipo "predice adidas en cualquier rectángulo".

## Solución propuesta

**Enfoque recomendado: oversampling físico en el training set** (más simple que tocar el dataloader).

En \`package_to_yolo\`, después del split:

\`\`\`python
target = max(class_counts.values())
threshold = target * 0.3  # cada clase al menos 30% del max

for cls, count in class_counts.items():
    if count < threshold:
        # Buscar items de train con esa clase
        items_with_cls = [i for i in train_indices if cls in valid_items[i].classes]
        n_extra = int(threshold) - count
        for k in range(n_extra):
            src = items_with_cls[k % len(items_with_cls)]
            # Copiar imagen+label con sufijo _augN
            shutil.copy2(src_img, dst_img.replace('.jpg', f'_aug{k}.jpg'))
\`\`\`

**IMPORTANTE**: solo duplicar en train/, NUNCA en val/. Si duplicás en val, el mAP miente.

## Alternativa

Class weights en el loss (Ultralytics no expone esto directamente — requiere subclase del trainer). Más limpio pero mucho más complejo.

## Criterios de aceptación

- [ ] Detección automática de desbalance: si max/min > 5, activar oversampling
- [ ] Reportar en response: `oversampled_classes: {bbva: 92, claro: 80}` (cuántas copias se hicieron)
- [ ] Asegurar que las imgs duplicadas tienen filenames únicos para que ultralytics no las cachee como duplicados
- [ ] No tocar val/

## Cuándo aplicar

Prioridad **baja** ahora (no tenés desbalance extremo). Activar cuando el ratio max:min sea >10:1. Esfuerzo: ~3h.
